src/binding.ts

```diff
@@ -62,74 +74,73 @@ export class BindingService {
         if (!connectionBindingsPerProjectKey.has(projectKey)) {
           connectionBindingsPerProjectKey.set(projectKey, []);
         }
-        connectionBindingsPerProjectKey.get(projectKey).push({ folder, binding });
+        connectionBindingsPerProjectKey.get(projectKey).push({folder, binding});
       }
     }
     return bindingsPerConnectionId;
   }
 
-  async createOrUpdateBinding(connectionId: string, contextValue: string) {
+  async createOrUpdateBinding(connectionId: string, contextValue: string, workspaceFolder?: VSCode.WorkspaceFolder) {
```

Comment:
This method is never called with the defined `workspaceFolder`. I think this argument should be removed and the method renamed to `createBinding`. OR, this method stays as it is and it is called instead of `editBinding`. (and `editBinding` is removed) 🧐